app/scripts/views/components/shareVirtualStudy/shareVirtualStudy.js

```diff
@@ -0,0 +1,206 @@
+      var previousSelectedCases = {};
+      if (this.showShareButton) {
+        $('.share-virtual-study').qtip({
+          style: {
+            classes: 'qtip-light qtip-rounded qtip-shadow'
+          },
+          show: {event: 'mouseover', ready: false},
+          hide: {fixed: true, delay: 200, event: 'mouseleave'},
+          position: {
+            my: 'bottom center',
+            at: 'top center',
+            viewport: $(window)
+          },
+          content: 'Share Virtual Study'
+        });
+        $('.share-cohort-btn').qtip({
+          style: {
+            classes: 'qtip-light qtip-rounded qtip-shadow ' +
+            'iviz-share-cohort-btn-qtip'
+          },
+          show: {event: 'click', ready: false},
+          hide: false,
+          position: {
+            my: 'top center',
+            at: 'bottom center',
+            viewport: $(window)
+          },
+          events: {
+            render: function(event, api) {
+              var tooltip = $('.iviz-share-cohort-btn-qtip .qtip-content');
+
+              tooltip.find('.share-cohort').click(function() {
+                tooltip.find('.shared').css('display', 'none');
+                tooltip.find('.dialog').css('display', 'none');
+                tooltip.find('.saving').css('display', 'none');
+                api.reposition();
+
+                // Copy virtual study link to clipboard
+                var $temp = $("<input>");
+                $("body").append($temp);
+                $temp.val(tooltip.find('.virtual-study-link').attr('href')).select();
+                document.execCommand("copy");
+                // Check if users copy url successfully
+                if ($temp.val() === tooltip.find('.virtual-study-link').attr('href')) {
+                  tooltip.find('.shared').css('display', 'block');
+                  tooltip.find('.dialog').css('display', 'none');
+                  api.reposition();
+                }
+                $temp.remove();
+              });
+
+            },
+            show: function() {
+              var tooltip = $('.iviz-share-cohort-btn-qtip .qtip-content');
+              tooltip.find('.dialog').css('display', 'block');
+              tooltip.find('.saving').css('display', 'block');
+              tooltip.find('.shared').css('display', 'none');
+
+              var cohortName = $('#study_name').val();
+              var cohortDescription = $('#study_desc').val();
+              if (_.isObject(vcSession)) {
+                self_.updateStats = true;
+
+                self_.$nextTick(function() {
+                  var saveCohort = false;
+
+                  if (_.isObject(self_.stats.selectedCases)) {
+                    var selectedCasesMap = {};
+                    _.each(self_.stats.selectedCases, function(study){
+                      selectedCasesMap[study.studyID] = study;
+                    });
+
+                    // When users double click "Share" button, the virtual cohort will 
```

Comment:
Why do we even allow user double click?

Reply:
Sorry, I didn't explain clear it clearly. Specifically, it's not double click. When a user clicks copy, it will trigger saving the current virtual cohort and return the url to the user. When a user want to see the cohort url, he/she needs to click `Share` button. We always show the url to user but we don't need to same virtual cohort every time if it is same with the previous saved cohort.